### PR TITLE
Add BeaconCache documentation

### PR DESCRIPTION
**Pull request by stefaneberl**

### CODE REVIEW

File: docs/internals.md [hunk 1]
```diff
@@ -65,13 +65,75 @@ retries.
...
+OpenKit instance via the builder the value can be set by calling `withBeaconCacheMaxRecordAge` with a long argument specifying the
+maximum record age in milliseconds. Bear in mind that Dynatrace/AppMon might still throw away the data, so setting
+this value too high might have no effect.
+
+It is possible to disable this strategy by setting the argument to `withBeaconCacheMaxRecordAge` to a value less than
+or equal to 0.
+
+#### Size based Eviction
+
+The second eviction strategy is used to keep the memory consumed by an OpenKit instance within certain boundaries.
```
Comment (thomasgrassauer) [inline docs/internals.md]: The second eviction .... -> The second eviction strategy is used to limit the memory consumption of OpenKit.

File: docs/internals.md [hunk 2]
```diff
@@ -65,13 +65,75 @@ retries.
...
+This strategy starts to run when the memory consumed by the BeaconCache exceeds a certain upper threshold and stops
```
Comment (thomasgrassauer) [inline docs/internals.md]: This strategy starts to run ... -> This strategy evicts beacons if the memory usage of the BeaconCache exceeds a configured upper bound and

File: docs/internals.md [hunk 3]
```diff
@@ -65,13 +65,75 @@ retries.
...
+as soon as the memory consumed by the cache reaches a configured lower bound. By default the upper boundary, when the
```
Comment (thomasgrassauer) [inline docs/internals.md]: By default the upper boundary ... -> By default, the upper boundary is set to 100 megabytes and the lower boundary is 80 megabytes.

File: docs/internals.md [hunk 4]
```diff
@@ -65,13 +65,75 @@ retries.
...
+
+An illustration how this cache looks like can be found below, followed by a detailed explanation.
+
+![diagram](./pics/OpenKit-BeaconCache.svg)
+
+### Basic BeaconCache Layout
+
+The BeaconCache is a 2-dimensional cache, storing data for each Session, internally also referred to
+as Beacon. Each OpenKit instance has such a cache, so if your application starts multiple OpenKit instances
+in parallel, keep in mind that each OpenKit instance consumes cache memory.
```
[the remark this answers is not in the document]
Comment (stefaneberl) [reply docs/internals.md]: As discussed, we'll leave this out here. "BeaconCache Eviction" mentions the second thread.

**Pull request merged**